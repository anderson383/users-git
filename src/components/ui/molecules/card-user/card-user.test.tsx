
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import CardUser from '.';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { User } from '@/services/repositories/github.repository';

const mockUser:User = {
  avatar_url: 'url image',
  login: 'test user',
  id: 'id-test'
}

const navigateMock = jest.fn()

jest.mock('react-router-dom', () => {
  return ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => jest.fn(),
    useNavigate: () => navigateMock
  }
  )
})


describe('Tests in Card User component', () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <CardUser user={mockUser}  />
      </BrowserRouter>
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should click card', async () => {
    setup();

    fireEvent.click(screen.getByTestId('card_user'));

    await act(() => Promise.resolve())

    expect(navigateMock).toHaveBeenCalled();
  });
});
