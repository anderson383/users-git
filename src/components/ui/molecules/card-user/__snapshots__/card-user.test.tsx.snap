// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Card User component Should render the component 1`] = `
<DocumentFragment>
  <div
    class="card_user"
    data-testid="card_user"
  >
    <div
      class="image"
    >
      <img
        alt=""
        src="url image"
      />
    </div>
    <div
      class="info"
    >
      <p>
        test user
      </p>
      <span>
        id-test
      </span>
    </div>
  </div>
</DocumentFragment>
`;
